go-controller/pkg/cluster/node.go

```diff
@@ -131,25 +133,68 @@ func (cluster *OvnClusterController) updateOvnNode(masterIP string,
 	return nil
 }
 
-// watchNamespaceUpdate starts watching namespace resources and calls back
-// the update handler logic if there is any namspace update event
-func (cluster *OvnClusterController) watchNamespaceUpdate(node *kapi.Node,
-	subnet string) error {
-	_, err := cluster.watchFactory.AddNamespaceHandler(
-		cache.ResourceEventHandlerFuncs{
-			UpdateFunc: func(old, newer interface{}) {
-				oldNs := old.(*kapi.Namespace)
-				oldMasterIP := oldNs.Annotations[MasterOverlayIP]
-				newNs := newer.(*kapi.Namespace)
-				newMasterIP := newNs.Annotations[MasterOverlayIP]
-				if newMasterIP != oldMasterIP {
-					err := cluster.updateOvnNode(newMasterIP, node, subnet)
-					if err != nil {
-						logrus.Errorf("Failed to update OVN node with new "+
-							"masterIP %s: %v", newMasterIP, err)
-					}
+func validateOVNConfigEndpoint(ep *kapi.Endpoints) bool {
+	if len(ep.Subsets) != 1 {
+		return false
+	}
+	if len(ep.Subsets[0].Ports) != 2 {
+		return false
+	}
+	return true
+
+}
+
+func updateOVNConfig(ep *kapi.Endpoints) {
+	if validateOVNConfigEndpoint(ep) {
```

Comment:
Here I'd just do

if !validateOVNConfigEndpoint(ep) {
    logrus.Errorf(...)
    return
}

other stuff

and save an indent